### PR TITLE
fix display of auth cards at all screen widths

### LINKED ISSUE #1267: Viewport a little too wide for iPhone SE
I'm not sure what the best solution is to this, but it looks like the viewport is too wide when using an iPhone SE. There's a little bit of horizontal scroll and some buttons go off screen. 

For example:

bs_realios_mobile_iphone se-11 0 2

### PR DESCRIPTION
**Pull request by aliceriot**
#### What are the relevant tickets?

closes #1267 

#### What's this PR do?

fixes the display for the auth cards at all page widths

#### How should this be manually tested?

Look at various auth pages and make sure that the display bug shown in the attached issue doesn't crop up anywhere. make sure you're testing in the chrome device simulator or a mobile device emulator.

### CODE REVIEW

File: static/scss/auth.scss [hunk 1]
```diff
@@ -15,12 +15,21 @@
...
+      margin: auto;
+      max-width: 470px;
+    }
 
     form,
-    .actions,
-    .card-contents {
+    .actions {
       width: 100%;
+      max-width: 350px;
```
[the remark this answers is not in the document]
Comment (aliceriot) [reply static/scss/auth.scss]: I think we don't need it on `.actions` (I looked at it's declared already below), but this happens if it's not set on `form`:

weird_withd

vs with it:

weird_withdfixed

but I think I can change it to have one `max-width` value for `form` at all display widths

File: static/scss/auth.scss [hunk 2]
```diff
@@ -15,12 +15,21 @@
...
+    }
+
+    .card-contents {
+      // this sets the width to 100% minus the padding
+      width: calc(100% - 36px);
```
[the remark this answers is not in the document]
Comment (aliceriot) [reply static/scss/auth.scss]: yeah, I think I inadvertently left that in from something I was trying initially

File: static/scss/auth.scss [hunk 3]
```diff
@@ -15,15 +15,18 @@
...
-    .card-contents {
-      width: 100%;
+    .main-content {
+      margin: auto;
+      max-width: 470px;
     }
   }
 
+  form {
+    max-width: 350px;
```
Comment (gsidebo) [inline static/scss/auth.scss]: I guess this is a nitpick, but I think the width should be fixed on `.card-contents`, not `form`. The 350px width is defined for `.card-contents` when the window is >800px wide. I think it should just be set to that width for any window width above mobile. That way you wouldn't have the `.card-contents` div suddenly growing to ~450px wide when the screen width drops below 800px. I don't think there's any real reason for that

ss 2018-09-27 at 17 27 24

Comment (aliceriot) [reply static/scss/auth.scss]: are you seeing the card contents div growing super wide like that? I can't repro that on my machine. The reason I set the the max-width on `form` was the mobile display, if I set the `max-width` on `.card-contents` instead it breaks the layout with flexbox:

max-width-on-card

there may be another way around it, but setting the max-width on the form and actions was the simplest way I found.

Comment (aliceriot) [reply static/scss/auth.scss]: so I can refactor it to set `max-width: 350px;` on `.card-contents` at all widths, and as long as I keep `margin: auto;` on at mobile the `width: 100%;` on `.form` and `.actions` will preserve the layout - I can do that if you think it's more logical.

Comment (Ferdi) [reply static/scss/auth.scss]: @aliceriot can you right-align the "next" button with the "email" box ?

Comment (aliceriot) [reply static/scss/auth.scss]: yeah, that's pretty simple

**Comment by aliceriot** [comment]
k should be fixed now

**Pull request merged**